server/test/unit/server/webservices/views/test_consumers.py

```diff
@@ -0,0 +1,1377 @@
+                                                     ConsumerContentActionView,
+                                                     ConsumerContentApplicabilityView,
+                                                     ConsumerContentApplicRegenerationView,
+                                                     ConsumerHistoryView, ConsumerProfilesView,
+                                                     ConsumerProfileResourceView,
+                                                     ConsumerResourceView,
+                                                     ConsumerResourceContentApplicRegenerationView,
+                                                     UnitInstallScheduleView,
+                                                     UnitInstallScheduleResourceView)
+
+
+class TestConsumersView(unittest.TestCase):
+    """
+    Test consumers view.
+    """
+    @mock.patch('pulp.server.webservices.views.consumers.serialization.binding.serialize')
+    @mock.patch('pulp.server.webservices.views.consumers.factory.consumer_bind_manager')
+    def test_expand_consumers(self, mock_factory, mock_serial):
+        """
+        Test for consumer info expansion with details/bindings
+        """
+        options = {'details': 'true'}
+        consumers_list = [{'id': 'c1'}]
+        bindings = [{'consumer_id': 'c1', 'repo_id': 'repo1', 'distributor_id': 'dist1'}]
+        mock_factory.return_value.find_by_criteria.return_value = bindings
+        mock_serial.return_value = {'consumer_id': 'c1', 'repo_id': 'repo1',
+                                    'distributor_id': 'dist1', '_href': '/some/c1/some_bind/'}
+
+        cons = consumers.expand_consumers(options, consumers_list)
+        expected_cons = [{'id': 'c1', 'bindings': [{'consumer_id': 'c1', 'repo_id': 'repo1',
+                         'distributor_id': 'dist1', '_href': '/some/c1/some_bind/'}]}]
+        self.assertEqual(cons, expected_cons)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.consumers.expand_consumers')
+    @mock.patch(
+        'pulp.server.webservices.views.consumers.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.consumers.factory.consumer_query_manager')
+    def test_get_all_consumers(self, mock_factory, mock_resp, mock_expand):
+        """
+        Test the consumers retrieval.
+        """
+        consumer_mock = mock.MagicMock()
+        resp = [{'id': 'foo', 'display_name': 'bar'}]
+        consumer_mock.find_all.return_value = resp
+        mock_factory.return_value = consumer_mock
+        mock_expand.return_value = resp
+
+        request = mock.MagicMock()
+        request.GET = {}
+        consumers = ConsumersView()
+        response = consumers.get(request)
+
+        expected_cont = [{'id': 'foo', 'display_name': 'bar', '_href': '/v2/consumers/foo/'}]
```

Comment:
I don't think this is a necessary change to make, but once you add tests for `add_link()` you can mock it in each of these functions.